pkg/prometheus/operator.go

```diff
@@ -2179,3 +2194,16 @@ func validateProberURL(url string) error {
 	}
 	return nil
 }
+
+func validateScrapeIntervalAndTimeout(p *monitoringv1.Prometheus, scrapeInterval, scrapeTimeout string) error {
+	if scrapeTimeout == "" {
+		return operator.CompareScrapeTimeoutToScrapeInterval(p.Spec.ScrapeTimeout, scrapeInterval)
```

Comment:
I modified it to handle the case

 `if the endpoint defines only the scrape interval value then compare it with the global scrape timeout (nothing to do if unset).`